lm_eval/models/huggingface.py

```diff
@@ -101,6 +101,8 @@ def __init__(
             self._device = self._model.device
             self._config = self._model.config
             gpus = 0
+            # default backend to causal if not specified
+            self.backend = self.backend if self.backend != "default" else "causal"
```

Comment:
Can we handle this case in `_get_backend()` and remove this line? 

It should be sufficient to simple settle for the existing behavior we had in that function (it defaults to causal if we can't detect both that our model is an HF model and that it is among the registered causal or seq2seq HF model types)